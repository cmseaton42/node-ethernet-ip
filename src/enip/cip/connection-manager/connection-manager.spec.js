const manager = require("./index");

describe("Connection Manager", () => {
    describe("Building", () => {
        it("Produces the Correct Output Buffer for ForwardOpen Request", () => {
            const { build_forwardOpen } = manager;
            const test = build_forwardOpen(10000);
            expect(test).toMatchSnapshot();
        });
        it("Produces the Correct Output Buffer for ForwardClose Request", () => {
            const { build_forwardClose } = manager;
            const test = build_forwardClose();
            expect(test).toMatchSnapshot();
        });
    });
    describe("Connection Parameters", () => {
        it("Produces the Correct Output number for connection parameters", () => {
            const { build_connectionParameters, owner, priority, fixedVar, connectionType } = manager;
            const test = build_connectionParameters(owner["Exclusive"], connectionType["PointToPoint"], priority["Low"], fixedVar["Variable"], 500);
            expect(test).toMatchSnapshot();
        });
        it("Error-cases: owner", () => {
            const { build_connectionParameters, priority, fixedVar, connectionType } = manager;
            expect(function() {
                build_connectionParameters("1000", connectionType["PointToPoint"], priority["Low"], fixedVar["Variable"], 500);
            }).toThrow();
        });
        it("Error-cases: connectionType", () => {
            const { build_connectionParameters, owner, priority, fixedVar } = manager;
            expect(function() {
                build_connectionParameters(owner["Exclusive"], "1000", priority["Low"], fixedVar["Variable"], 500);
            }).toThrow();
        });
        it("Error-cases: priority", () => {
            const { build_connectionParameters, owner, fixedVar, connectionType } = manager;
            expect(function() {
                build_connectionParameters(owner["Exclusive"], connectionType["PointToPoint"], "1000", fixedVar["Variable"], 500);
            }).toThrow();
        });
        it("Error-cases: fixedVar", () => {
            const { build_connectionParameters, owner, priority, connectionType } = manager;
            expect(function() {
                build_connectionParameters(owner["Exclusive"], connectionType["PointToPoint"], priority["Low"], "1000", 500);
            }).toThrow();
        });
        it("Error-cases: size", () => {
            const { build_connectionParameters, owner, priority, fixedVar, connectionType } = manager;
            expect(function() {
                build_connectionParameters(owner["Exclusive"], connectionType["PointToPoint"], priority["Low"], fixedVar["Variable"], 999999);
            }).toThrow();
        });
    });
});
