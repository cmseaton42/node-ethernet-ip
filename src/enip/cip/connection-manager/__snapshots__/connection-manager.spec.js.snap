// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection Manager Building Produces the Correct Output Buffer for ForwardClose Request 1`] = `
Object {
  "data": Array [
    3,
    125,
    66,
    66,
    51,
    51,
    55,
    19,
    0,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Connection Manager Building Produces the Correct Output Buffer for ForwardOpen Request 1`] = `
Object {
  "data": Array [
    3,
    125,
    17,
    17,
    17,
    17,
    34,
    34,
    34,
    34,
    66,
    66,
    51,
    51,
    55,
    19,
    0,
    0,
    3,
    0,
    0,
    0,
    16,
    39,
    0,
    0,
    244,
    67,
    16,
    39,
    0,
    0,
    244,
    67,
    163,
  ],
  "type": "Buffer",
}
`;

exports[`Connection Manager Connection Parameters Produces the Correct Output number for connection parameters 1`] = `17396`;
